### PR TITLE
Topics: Get all articles by topic

### PR DESCRIPTION
**Pull request by Drapegnik**
- topics api - http://api.wir.by/api/topics/articles/times
- refactor `initAll` script
- add `mapIds`, `getId`, `getTagsByTopic` and `getArticlesByTag` helpers

### CODE REVIEW

**Comment by Drapegnik** [comment]
[topics page api](https://trello.com/c/86zM93lN/367-topics-page-api)

File: src/utils/getters.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  const topicsById = keyBy(topics, '_id');
+  return tags.reduce((acc, { topic, _id }) => {
+    const topicSlug = topicsById[topic].slug;
+    acc[topicSlug] = acc[topicSlug] || [];
+    acc[topicSlug].push(_id);
+    return acc;
+  }, {});
+};
+
+// when articles didn't populated with tags - use `tagsById` to provide `slugs`
```
Comment (UladBohdan) [inline src/utils/getters.js]: *were not populated

File: .eslintrc.js [hunk 2]
```diff
@@ -21,8 +21,12 @@ module.exports = {
     'mocha/no-skipped-tests': 'error',
     'mocha/no-synchronous-tests': 'error',
     'mocha/no-top-level-hooks': 'error',
-    'mocha/valid-suite-description': ['warn', '^[A-Z#]'],
-    'mocha/valid-test-description': 'warn',
+    'mocha/valid-suite-description': ['warn', /^[A-Z#]/],
```
Comment (UladBohdan) [inline .eslintrc.js]: What are these changes for?

File: src/api/tag/tests.js [hunk 3]
```diff
@@ -211,4 +210,17 @@ describe('Tags API', () => {
...
+  it('should return articles by topic', () =>
+    request
+      .get('/api/topics/articles/locations')
+      .expect(HttpStatus.OK)
+      .expect(({ body: { articles, articlesByTag, tags, topic } }) => {
+        expect(topic.slug).to.equal('locations');
+        expect(articles).to.have.length(2);
+        expect(tags).to.have.length(1);
+        expect(tags[0].slug).to.equal('miensk');
+        expect(Object.keys(articlesByTag)).to.eql(['miensk']);
```
Comment (UladBohdan) [inline src/api/tag/tests.js]: `eql` - is it a new shortcut?

File: src/utils/getters.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import keyBy from 'lodash/keyBy';
+
+export const getId = ({ _id }) => _id.toString();
+
+export const safeGetId = o => o._id || o;
```
Comment (UladBohdan) [inline src/utils/getters.js]: I can't see where you're using it apart from the tests.

File: src/api/tag/tests.js (hunk 3 above)
Comment (Drapegnik) [reply src/api/tag/tests.js]: https://www.chaijs.com/api/bdd/#method_eql

File: src/utils/getters.js (hunk 4 above)
Comment (Drapegnik) [reply src/utils/getters.js]: hm, it's gone after refactoring :)

File: .eslintrc.js (hunk 2 above)
Comment (Drapegnik) [reply .eslintrc.js]: for suit cases message:
```js
it('[getId] should return string'
```

Comment (Drapegnik) [reply .eslintrc.js]: for consistency - `/ /` - is regexp expression

**Pull request merged**